fsspec/implementations/local.py

```diff
@@ -46,31 +46,43 @@ def rmdir(self, path):
 
     def ls(self, path, detail=False, **kwargs):
         path = self._strip_protocol(path)
-        paths = [posixpath.join(path, f) for f in os.listdir(path)]
         if detail:
-            return [self.info(f) for f in paths]
+            return [self.info(f) for f in os.scandir(path)]
         else:
-            return paths
+            return [posixpath.join(path, f) for f in os.listdir(path)]
 
     def glob(self, path, **kwargs):
         path = self._strip_protocol(path)
         return super().glob(path, **kwargs)
 
     def info(self, path, **kwargs):
-        path = self._strip_protocol(path)
-        out = os.stat(path, follow_symlinks=False)
-        if os.path.isdir(path):
-            t = "directory"
-        elif os.path.isfile(path):
-            t = "file"
+        if isinstance(path, str):
+            path = self._strip_protocol(path)
+            out = os.stat(path, follow_symlinks=False)
+            link = os.path.islink(path)
+            if os.path.isdir(path):
+                t = "directory"
+            elif os.path.isfile(path):
+                t = "file"
+            else:
+                t = "other"
         else:
-            t = "other"
+            # scandir DirEntry
+            out = path.stat(follow_symlinks=False)
```

Comment:
is this a breaking change in the API? in pytorch lightning tests we're seeing this stacktrace now: 
```
kwargs = {}
    def info(self, path, **kwargs):
        if isinstance(path, str):
            path = self._strip_protocol(path)
            out = os.stat(path, follow_symlinks=False)
            link = os.path.islink(path)
            if os.path.isdir(path):
                t = "directory"
            elif os.path.isfile(path):
                t = "file"
            else:
                t = "other"
        else:
            # scandir DirEntry
>           out = path.stat(follow_symlinks=False)
E           TypeError: stat() got an unexpected keyword argument 'follow_symlinks'
```
example: https://github.com/PyTorchLightning/pytorch-lightning/runs/2766424283?check_suite_focus=true

Reply:
No the API did not change, this path is only meant to be called by internal functions. The test is whether the path is a str, and I suspect that the tmpdir fixture used for the test is not actually a str, but something str-like. `str(tempdir)` in the test would solve it. However, fsspec maybe should have coped with this in the more user-facing `isdir/info` functions (if my guess about the cause is correct).

Reply:
yes, you're right. `tmpdir` is a py.path.local object. while we can update our tests to handle with this, does it make sense to update the instance check as well?